sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -45,47 +46,60 @@ func init() {
 }
 
 func main() {
-	os.Exit(_main())
-}
-
-func _main() int {
 	// Parse command-line arguments
 	var (
 		containerID      string
 		source           string
 		containerdSocket string
 		namespace        string
 		superpowered     bool
-		pullImageOnly    bool
 	)
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "usage: %s run|pull-image\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+	fs := flag.NewFlagSet("", flag.ExitOnError)
+	fs.StringVar(&containerdSocket, "containerd-socket", "/run/host-containerd/containerd.sock", "specifies the path to the containerd socket")
+	fs.StringVar(&namespace, "namespace", "default", "specifies the containerd namespace")
+	fs.StringVar(&source, "source", "", "the image to be pulled")
+	fs.BoolVar(&superpowered, "superpowered", false, "specifies whether to create the container with `superpowered` privileges or not")
+	fs.StringVar(&containerID, "container-id", "", "the id of the container to manage")
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "expected one of the following subcommands: 'run', 'pull-image'")
+		flag.Usage()
+		os.Exit(2)
+	}
+	fs.Parse(os.Args[2:])
+
+	var subcommand = os.Args[1]
+	switch subcommand {
+	case "run":
+		os.Exit(runCtr(containerdSocket, namespace, containerID, source, superpowered))
+	case "pull-image":
+		os.Exit(pullImageOnly(containerdSocket, namespace, source))
+	default:
+		fmt.Errorf("expected one of the following subcommands: 'run', 'pull-image'")
```

Comment:
`fmt.Errorf` isn't the right approach here; that will just return an `error` but since it's not assigned it gets dropped on the floor.

`fmt.Fprintf` is what you were looking for here, but probably a better pattern is similar to what you had prior to this refactoring change:

```go
func main() {
	if err := _main(); err != nil {
		fmt.Fprintln(err)
		os.Exit(1)
	}
}

func _main() error {
	// main body here
}
```